wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Message Details/MessageDetailsContentViewController.swift

```diff
@@ -254,28 +282,73 @@ final class MessageDetailsContentViewController: UIViewController {
 extension MessageDetailsContentViewController: UICollectionViewDataSource, UICollectionViewDelegateFlowLayout {
 
     func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
-        return cells.count
+        return sections[section].items.count
+    }
+
+    func numberOfSections(in collectionView: UICollectionView) -> Int {
+        return sections.count
+    }
+
+    func collectionView(
+        _ collectionView: UICollectionView,
+        layout collectionViewLayout: UICollectionViewLayout,
+        referenceSizeForHeaderInSection section: Int
+    ) -> CGSize {
+        header.titleLabel.text = "headerText"
```

Comment:
to delete?

Reply:
If we remove that, then we have no header text for our section header. What I can do is to leave it as an empty string!? Do you have any better alternatives?

Reply:
That's also fixed with my latest commits.